app/models/journal/work_package_journal.rb

```diff
@@ -32,4 +32,10 @@ class Journal::WorkPackageJournal < Journal::BaseJournal
   belongs_to :project
   belongs_to :assigned_to, class_name: 'Principal'
   belongs_to :responsible, class_name: 'Principal'
+
+  def journaled_attributes
+    journaled = super
+    journaled.delete(:duration) unless OpenProject::FeatureDecisions.work_packages_duration_field_active?
+    journaled
```

Comment:
This will result in the duration not being journaled at all. So while the feature flag is off, no value changes are noted in the journals. This will make enabling the feature later on harder as the information about what happened before activating the flag just does not exist.

Is it possible to rather continue journalizing the property but prevent it from being rendered? Since there is now a custom renderer for the duration, can it return `nil` in its `render` method if the feature flag is disabled? It doesn't look like the [`JournalFormatter`](https://github.com/opf/openproject/blob/dev/lib_static/plugins/acts_as_journalized/lib/journal_formatter.rb#L110) is fit for that yet, but it in some cases already returns `nil` so its callers should be fine with doing that as well.

Reply:
I think the `JournalChanges` module is more suitable, because it will also hide the duration from the work package pdf export as well. WDYT?

Reply:
Sounds good